packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/CoinControl/UtxoSearch.tsx

```diff
@@ -0,0 +1,82 @@
+import { useCallback, useRef, Dispatch, SetStateAction, ChangeEvent, KeyboardEvent } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { Input, Icon, KEYBOARD_CODE } from '@trezor/components';
+import { useTranslation } from 'src/hooks/suite/useTranslation';
+import { isFeatureFlagEnabled } from '@suite-common/suite-utils';
+import { borders, spacingsPx } from '@trezor/theme';
+
+const Container = styled.div`
+    display: flex;
+    align-items: center;
+    width: 100%;
+`;
+
+const StyledInput = styled(Input)<{ $isExpanded: boolean }>`
+    width: 100%;
+    margin-right: ${spacingsPx.xs};
+    border-radius: ${borders.radii.full};
+
+    input {
+        height: 38px;
+        width: 100%;
+        border: none;
+    }
+`;
+
+const SearchIcon = styled(Icon)`
+    cursor: pointer;
+`;
+
+export interface SearchProps {
+    searchQuery: string;
+    setSearch: Dispatch<SetStateAction<string>>;
+    setSelectedPage: Dispatch<SetStateAction<number>>;
+}
+
+export const UtxoSearch = ({ searchQuery, setSearch, setSelectedPage }: SearchProps) => {
+    const theme = useTheme();
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const { translationString } = useTranslation();
+
+    const onKeyDown = useCallback(
+        (event: KeyboardEvent) => {
+            // Handle ESC (un-focus)
+            if (event.code === KEYBOARD_CODE.ESCAPE && inputRef.current) {
+                setSearch('');
+                inputRef.current.blur();
+            }
+        },
+        [setSearch],
+    );
+
+    const onSearch = useCallback(
+        async ({ target }: ChangeEvent<HTMLInputElement>) => {
+            setSearch(target.value);
+            setSelectedPage(1);
+        },
+        [setSearch],
+    );
+
+    if (!isFeatureFlagEnabled('SEARCH_TRANSACTIONS')) {
+        return null;
+    }
+
+    return (
+        <Container>
+            <StyledInput
+                $isExpanded
+                data-test="@wallet/accounts/search-icon"
+                innerRef={inputRef}
+                innerAddon={<SearchIcon icon="SEARCH" size={16} color={theme.iconSubdued} />}
+                placeholder={translationString('TR_SEARCH_TRANSACTIONS')}
```

Comment:
we need a different translation string here

Reply:
Just to confirm. I have just added a new translation string `TR_SEARCH_UTXOS` in support/messages.ts. We don't need to update any lang json file manually right?

Reply:
Yes, all you need is to add the new token to the `packages/suite/src/support/messages.ts`. All languages are then generated automatically (with help of our translators).